pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go

```diff
@@ -490,7 +493,24 @@ func (m *kubeapi) Count(ctx context.Context, entity datastore.Entity, filterOpti
 }
 
 func verifyValue(v string) string {
+
 	s := strings.ReplaceAll(v, "@", "-")
 	s = strings.ReplaceAll(s, " ", "-")
 	return strings.ToLower(s)
 }
+
+func convertIndex2Labels(index map[string]interface{}) map[string]string {
+	if index == nil {
+		return nil
+	}
+	ret := make(map[string]string, len(index))
+	for k, v := range index {
+		value := pkgUtils.ToString(v)
+		if value == "" {
+			klog.Warningf("unable to cast %#v of type %T to string", v, v)
+			continue
+		}
+		ret[k] = pkgUtils.ToString(v)
```

Comment:
need check the length of Key and Value, make sure there length is less then K8S restriction.
If Key or Value is too long, what is solution?

Reply:
Index will convert to k8s labels, k8s labels has len limited.The current version of vela does nothing,If necessary can open a feat or fix separately to solve the problem